### DISCUSSION TITLE
Multiple Cognito Providers

### DISCUSSION THREAD

**Discussion by NicoChaima72** [nextauthjs/next-auth#9581] [Help]
“Hello, I am trying to configure 2 cognito providers for the application, according to the documentation and forums, it would be enough to declare them and assign them a different id, but when I do that, the second provider gives me an error, if I assign a custom id to a provider, I always get that error, does anyone know how to configure it correctly?”

`
[...nextauth].ts
adapter: PrismaAdapter(prisma),
  providers: [
    CognitoProvider({
      clientId: process.env.COGNITO_CLIENT_ID as string,
      clientSecret: process.env.COGNITO_CLIENT_SECRET as string,
      issuer: process.env.COGNITO_ISSUER as string,
    }),
    CognitoProvider({
      clientId: process.env.COGNITO_CLIENT_ID_2 as string,
      clientSecret: process.env.COGNITO_CLIENT_SECRET_2 as string,
      issuer: process.env.COGNITO_ISSUER_1 as string,
    }),
  ]`

`page.tsx
<Button
              variant="gradient"
              size="xl"
              gradient={{ from: "orange", to: "red" }}
              onClick={() => signIn("cognito")}
            >
              Iniciar sesión
            </Button>
            <Button
              variant="gradient"
              size="xl"
              gradient={{ from: "orange", to: "red" }}
              onClick={() => signIn("cognito2")}
            >
              Iniciar sesión 2
            </Button>`

<img width="679" alt="Captura de pantalla 2024-01-08 a la(s) 14 15 35" src="https://github.com/nextauthjs/next-auth/assets/26178362/cfa5269c-77cb-460f-8d11-abcd230df862">
<img width="647" alt="Captura de pantalla 2024-01-08 a la(s) 14 16 49" src="https://github.com/nextauthjs/next-auth/assets/26178362/08be47f1-f7d6-4485-b768-1c2205142800">

![Captura de pantalla 2024-01-08 a la(s) 14 18 39](https://github.com/nextauthjs/next-auth/assets/26178362/48be495e-5c3a-426d-a2b1-a841038e8904)
![Captura de pantalla 2024-01-08 a la(s) 14 54 50](https://github.com/nextauthjs/next-auth/assets/26178362/3133cf33-a9f0-406e-8bb1-f4b06e6f55fd)

**Comment by Sergioeabarcaf** [comment]
@NicoChaima72 , I see that your problem is that you are using the 'id' when calling the [signIn](https://next-auth.js.org/getting-started/client#signin) function, but this function receives providers (and options). Therefore, you could use provider values like "`azure`", "`google`", "`email`"...

I am trying to do the same, having two Cognito, but it seems it is not possible :/

I managed to connect two providers, but they have to be different. In my case, I used Cognito and a credentials provider.
